common/flatpak-dir.c

```diff
@@ -11970,8 +12019,25 @@ get_locale_langs_from_accounts_dbus (GDBusProxy *proxy)
         }
     }
 
-  if (langs->len == 0)
-    return NULL; /* No defined languages, fall back to all languages */
+  /* At least one user with no defined language, fall back to system default
```

Comment:
This logic seems un-necessarily contrived. I think it would be a little simpler to just take the system defaults, then add any user defaults to them if they differ, and then return NULL if that gets us nothing. The overwhelmingly likely case is that the system default applies to most if not all of the users, so we're hardly going to be un-necessarily pulling stuff in. In any case, if a new user is added, they will at least have the default locale represented right away.

Reply:
Do the performance implications of that worry you? In other words, doing two D-Bus calls unconditionally vs. doing one and only doing the other if necessary.

Reply:
D-Bus calls are not fast. I’d prefer to avoid an unnecessary one if possible. Might be worth pointing out in a code comment that that’s why the code is ‘contrived’.

Reply:
Including the default system locale seems worthwhile in any case, doesn't it?

Reply:
We're currently doing N+1 D-Bus calls for N cached users on the system. This would just be N+2. Not the biggest fish to fry I don't think.

Reply:
Certainly not the biggest fish to fry, but when `N` is typically 1 or 2, that extra call becomes a bit more significant. I’m not going to bikeshed over this though.